CDTDatastore/CDTReplicator/CDTReplicator.m

```diff
@@ -44,12 +44,44 @@ @interface CDTReplicator ()
 @property (nonatomic, readwrite) NSInteger changesProcessed;
 @property (nonatomic, readwrite) NSInteger changesTotal;
 @property (nonatomic, readwrite) NSError *error;
+@property (nonatomic, nullable, readwrite) CDTReplicator* retainedSelf;
 
 @property (nonatomic, copy) CDTFilterBlock pushFilter;
 @property (nonatomic) BOOL started;
 
 @end
-
+/*
+    The CDTReplicator class is "fire and forget". This is so that an application
+    can choose not to maintain a reference to an instance of CDTReplicator
+    without the instance being immediately deallocated (and so failing to complete
+    its replication).
+ 
+    This is implemented by each instance retaining itself by assigning to its
+    own retainedSelf property when replication is started. Care must be taken to ensure
+    that this extra retain is released when the replication is complete.
+ 
+    This is implemented by retaining and releasing at the points noted in
+    the lifecycle diagram below:
+  
+    -startWithGroup:Error: -> start TDReplicator -> retain self -> return
```

Comment:
I'm not convinced that this is really a good 'lifecycle diagram' - I'll see if I can suggest an alternative.